### PR TITLE
feat(Dropdown): selectedIndex -> selectedId

### PR DESCRIPTION
**Pull request by KoichiKiyokawa**
part of #1003

Modify `selectedIndex` to `selectedId` in Dropdown component.
(I will create another PR for ComboBox later to make the diff smaller.)

### CODE REVIEW

File: src/Dropdown/Dropdown.svelte [hunk 1]
```diff
@@ -123,12 +125,12 @@
     highlightedIndex = index;
   }
 
-  $: if (selectedIndex > -1) {
+  $: if (selectedId !== undefined) {
     dispatch("select", { selectedId, selectedIndex, selectedItem });
```
Comment (KoichiKiyokawa) [inline src/Dropdown/Dropdown.svelte:129]: I'm not sure if we should dispatch `selectedIndex` 🤔

Comment (metonym) [reply src/Dropdown/Dropdown.svelte:129]: Let's leave it in for now

**Review by metonym — CHANGES_REQUESTED** [review]
This looks great!

I agree that `selectedIndex` can be omitted from the `select` event.

File: src/Dropdown/Dropdown.svelte (hunk 1 above)
Comment (KoichiKiyokawa) [reply src/Dropdown/Dropdown.svelte]: Done in d52828a 🙆‍♂️

**Review by metonym — APPROVED** [review]
Excellent work! I tested two-way binding and the `on:select` worked as expected.

I'd like to get this merged/released in the next minor version (0.53) along with the changes to ComboBox.

**Comment by KoichiKiyokawa** [comment]
@metonym 
Thank you for reviewing :tada:
I created the PR for `ComboBox` in #1016

**Comment by metonym** [comment]
Released in [v0.53.0](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.53.0).

**Pull request merged**